src/Storages/ObjectStorage/DataLakes/DeltaLake/KernelHelper.cpp

```diff
@@ -0,0 +1,115 @@
+#include "config.h"
+
+#if USE_DELTA_KERNEL_RS
+#include <Storages/ObjectStorage/S3/Configuration.h>
+#include "KernelHelper.h"
+#include "KernelUtils.h"
+
+namespace DB::ErrorCodes
+{
+    extern const int NOT_IMPLEMENTED;
+}
+
+namespace DeltaLake
+{
+
+/// A helper class to manage S3-compatible storage types.
+class S3KernelHelper final : public IKernelHelper
+{
+public:
+    S3KernelHelper(
+        const DB::S3::URI & url_,
+        const std::string & access_key_id_,
+        const std::string & secret_access_key_,
+        const std::string & region_)
+        : url(url_)
+        , access_key_id(access_key_id_)
+        , secret_access_key(secret_access_key_)
+        , region(region_)
+        , table_location(getTableLocation(url_))
+    {
+    }
+
+    const std::string & getTableLocation() const override { return table_location; }
+
+    const std::string & getDataPath() const override { return url.key; }
+
+    ffi::EngineBuilder * createBuilder() const override
```

Comment:
Should we care about memory for struct `EngineBuilder`? Or it's managed by rust?

Reply:
There is no deleter function provided in ffi api, so it must mean that we do not need to manage it, I think it is managed by the engine created next line